src/main/java/graphql/execution/SubscriptionExecutionStrategy.java

```diff
@@ -77,7 +78,13 @@ public CompletableFuture<ExecutionResult> execute(ExecutionContext executionCont
             }
             Function<Object, CompletionStage<ExecutionResult>> mapperFunction = eventPayload -> executeSubscriptionEvent(executionContext, parameters, eventPayload);
             boolean keepOrdered = keepOrdered(executionContext.getGraphQLContext());
-            SubscriptionPublisher mapSourceToResponse = new SubscriptionPublisher(publisher, mapperFunction, keepOrdered);
+
+            InstrumentationReactiveResultsParameters instrumentationReactiveResultsParameters = new InstrumentationReactiveResultsParameters(executionContext, InstrumentationReactiveResultsParameters.ResultType.SUBSCRIPTION);
+            InstrumentationContext<Void> reactiveCtx = nonNullCtx(executionContext.getInstrumentation().beginReactiveResults(instrumentationReactiveResultsParameters, executionContext.getInstrumentationState()));
+            reactiveCtx.onDispatched();
+
+            SubscriptionPublisher mapSourceToResponse = new SubscriptionPublisher(publisher, mapperFunction, keepOrdered,
+                    throwable -> reactiveCtx.onCompleted(null, throwable));
```

Comment:
we pass in the call back here.  Why?

We want the SubscriptionPublisher to be the implementation we pass back - we have tests for this.

So I changed SubscriptionPublisher internally to know when its finished